constant/skill/class.go

```diff
@@ -17,3 +17,277 @@ type gm struct {
+	JobID        int
+	Meditation   Skill
+	MpEater      Skill
+	PoisonBreath Skill
+	FireArrow    Skill
+	Slow         Skill
+}
+
+type FPMage struct {
+	JobID                int
+	ElementAmplification Skill
+	ElementComposition   Skill
+	PartialResistance    Skill
+	PoisonMyst           Skill
+	Seal                 Skill
+	SpellBooster         Skill
+	Explosion            Skill
+}
+
+type ILWizard struct {
+	JobID       int
+	ColdBeam    Skill
+	Meditation  Skill
+	MpEater     Skill
+	Slow        Skill
+	ThunderBolt Skill
+}
+
+type ILMage struct {
+	JobID                int
+	ElementAmplification Skill
+	ElementComposition   Skill
+	IceStrike            Skill
+	PartialResistance    Skill
+	Seal                 Skill
+	SpellBooster         Skill
+	Lightening           Skill
+}
+
+type Cleric struct {
+	JobID      int
+	Bless      Skill
+	Heal       Skill
+	Invincible Skill
+	MpEater    Skill
+	HolyArrow  Skill
+}
+
+type Priest struct {
+	JobID               int
+	Dispel              Skill
+	Doom                Skill
+	ElementalResistance Skill
+	HolySymbol          Skill
+	MysticDoor          Skill
+	SummonDragon        Skill
+}
+
+type Archer struct {
+	JobID            int
+	BlessingOfAmazon Skill
+	CriticalShot     Skill
+	Focus            Skill
+}
+
+type Hunter struct {
+	JobID          int
+	PowerKnockback Skill
+	ArrowBomb      Skill
+	BowBooster     Skill
+	BowMastery     Skill
+	SoulArrow      Skill
+}
+
+type Ranger struct {
+	JobID      int
+	MortalBlow Skill
+	Puppet     Skill
+	SilverHawk Skill
+}
+
+type Crossbowman struct {
+	JobID           int
+	PowerKnockback  Skill
+	CrossbowBooster Skill
+	CrossbowMastery Skill
+	SoulArrow       Skill
+}
+
+type Sniper struct {
+	JobID       int
+	Blizzard    Skill
+	GoldenEagle Skill
+	MortalBlow  Skill
+	Puppet      Skill
+}
+
+type Rogue struct {
+	JobID      int
+	NimbleBody Skill
+	DarkSight  Skill
+	Disorder   Skill
+	DoubleStab Skill
+	LuckySeven Skill
+}
+
+type Assassin struct {
+	JobID         int
+	ClawBooster   Skill
+	ClawMastery   Skill
+	CriticalThrow Skill
+	Endure        Skill
+	Drain         Skill
+	Haste         Skill
+}
+
+type Hermit struct {
+	JobID         int
+	Alchemist     Skill
+	Avenger       Skill
+	MesoUp        Skill
+	ShadowMeso    Skill
+	ShadowPartner Skill
+	ShadowWeb     Skill
+}
+
+type Bandit struct {
+	JobID         int
+	DaggerBooster Skill
+	DaggerMastery Skill
+	Endure        Skill
+	Haste         Skill
+	SavageBlow    Skill
+	Steal         Skill
+}
+
+type ChiefBandit struct {
+	JobID         int
+	Assaulter     Skill
+	BandOfThieves Skill
+	Chakra        Skill
+	MesoExplosion Skill
+	MesoGuard     Skill
+	Pickpocket    Skill
+}
+
+type GM struct {
+	JobID           int
+	Bless           Skill
+	Haste           Skill
+	HealPlusDispell Skill
+	Hide            Skill
+	HolySymbol      Skill
+	Resurrection    Skill
+	SuperDragonRoar Skill
+	Teleport        Skill
+
+	ItemExplosion Skill
+	ShadowPartner Skill
+	JumpDown      Skill
+}
+
+var (
```

Comment:
I believe it would be better to have the variables as global vars and simply init them in the package. 

Ex.

```
var Swordsman swordsman
func init() {
    Swordsman = swordsman{
      JobID: WarriorJobID,
      ImprovedMaxHpIncrease:  ImprovedMaxHpIncrease,
      etc..
    }
}
```

Reply:
Yea that makes sense. 

I'm not exactly sure how the class skills will end up being used - I wrote the file near the start of the project a while ago. Do you have any plans on how to use these structs?

Reply:
I believe they will really just be used as conditionals in an easy to read format. They might be of use in other ways, but I haven't thought of others yet. 

Example:

```
if ( Assassin.Haste || Bandit.Haste) {
 Increase Speed
}
```